syntax = "proto3";

option java_multiple_files = true;
option java_package = "ds.adam.VideoDoorbell";
option java_outer_classname = "VideoDoorbellServiceImpl";

package VideoDoorbell;

service VideoDoorbellService{

	//2nd
	rpc homeVideo(stream Video) returns (stream BellRequest){}

	//3rd
	rpc doorStatus(stream DoorLockStatus) returns(DoorLockTamper){}

}

message Video{

	int32 exampleNumber1 = 1;

}

message BellRequest{

	int32 exampleNumber2 = 2;

	enum Visual{
		GREEN = 0;
		RED = 2;
		ERROR = 3;
	}

	Visual visual = 3;
}

message DoorLockStatus{

	bool check = 1;

}

message DoorLockTamper{

	bool intruder = 1;

}
